R/hook.R

```diff
@@ -72,3 +72,46 @@ session_load_hook <- function(user_hook = NULL) {
 
   paste0(deparse(hook), "\n")
 }
+
+user_hooks_env <- new.env(parent = emptyenv())
+
+#' Add a user hook to be executed before launching an R subprocess
+#'
+#' This function allows users of `callr` to specify functions that get invoked
+#' whenever an R session is launched. The function can modify the environment
+#' variables and command line arguments.
+#'
+#' The prototype of the hook function is `function (options)`, and it is
+#' expected to return the modified `options`.
+#' @param ... Named argument specifying a hook function to add, or `NULL` to
+#'   delete the named hook.
+#' @return `add_hook` is called for its side-effects.
+#' @export
+
+add_hook <- function(...) {
+  if (...length() != 1L) {
```

Comment:
callr currently supports R >= 3.4 and `...length()` is only available from R 3.5.x. I can fix this, though.